### PR TITLE
RTL Conversion for NxButton - RSC-1228

### PR DESCRIPTION
**Pull request by cindyko226**
https://issues.sonatype.org/browse/RSC-1228

### CODE REVIEW

File: lib/src/components/NxButton/__tests__/NxButton.test.tsx [hunk 1]
```diff
@@ -5,82 +5,66 @@
...
-    expect(titleNoContext).toHaveProp('title', 'bar');
+    await waitFor(() => expect(iconOnlyBtnNoTitle).not.toHaveAccessibleName());
+    await waitFor(() => expect(iconOnlyBtnTitle).toHaveAccessibleName('Alert'));
   });
 
-  it('throws an error when it contains both disabled and title props', function() {
-    expect(() => {
-      shallow(<NxButton variant="icon-only" title="Delete" disabled><NxFontAwesomeIcon icon={faTrashAlt}/></NxButton>);
-    }).toThrow(TypeError);
+  it('throws an error when it contains both disabled and title props', async function() {
```
Comment (EmilyBermudez88) [inline lib/src/components/NxButton/__tests__/NxButton.test.tsx:66]: Is it worth testing here that using `className='disabled'` with a title prop doesn't throw an error?

Comment (cindyko226) [reply lib/src/components/NxButton/__tests__/NxButton.test.tsx:66]: Updated here 5926ddaecd1bd00b7e1df5dbcc665c1b5c83268b.

File: lib/src/components/NxButton/__tests__/NxButton.test.tsx [hunk 2]
```diff
@@ -5,82 +5,68 @@
...
 
 describe('NxButton', function() {
-  it('renders a button', function() {
-    const button = shallow(<NxButton />);
+  const quickRender = rtlRender(NxButton, {}),
+      renderEl = rtlRenderElement(NxButton, {});
 
-    expect(button).toHaveClassName('nx-btn');
+  it('renders a button', function() {
+    expect(quickRender().getByRole('button').tagName).toBe('BUTTON');
```
Comment (rpokorny) [inline lib/src/components/NxButton/__tests__/NxButton.test.tsx:18]: In this case I think this is an implementation detail that doesn't need to be tested

File: lib/src/components/NxButton/__tests__/NxButton.test.tsx [hunk 3]
```diff
@@ -5,82 +5,68 @@
...
+    expect(quickRender({ disabled: undefined }).getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: false }).getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: true }).getByRole('button')).toBeDisabled();
   });
 
   it('disabled by class button has aria-disabled true', function() {
-    const button = shallow(<NxButton className="disabled">Disabled Button</NxButton>);
-
-    expect(button).toHaveProp('aria-disabled', true);
+    expect(quickRender().getByRole('button')).not.toBeDisabled();
```
Comment (rpokorny) [inline lib/src/components/NxButton/__tests__/NxButton.test.tsx:54]: I assume this was supposed to be
```suggestion
    expect(quickRender({ className: 'disabled' }).getByRole('button')).not.toBeDisabled();
```

**Review by rpokorny — APPROVED** [review]

File: lib/src/components/NxButton/__tests__/NxButton.test.tsx (hunk 2 above)
Comment (cindyko226) [reply lib/src/components/NxButton/__tests__/NxButton.test.tsx]: Removed here 86e0e240ca7e34829756a1c515c2ed3d645221dd.

**Pull request merged**